crates/datafusion-optimizer-rules/src/physical_plan/duckdb/intermediate_index_cte.rs

```diff
@@ -11,41 +12,37 @@ use datafusion::physical_plan::ExecutionPlan;
 use datafusion::sql::sqlparser::ast::helpers::attached_token::AttachedToken;
 use datafusion::sql::sqlparser::ast::{
     BinaryOperator, Cte, Expr, Ident, ObjectNamePart, Select, SelectItem, SetExpr, Statement,
-    TableAlias, TableFactor, TableWithJoins, Value, ValueWithSpan, With, visit_expressions,
-    visit_expressions_mut, visit_relations,
+    TableAlias, TableFactor, TableWithJoins, Value, ValueWithSpan, VisitMut, With,
+    visit_expressions, visit_expressions_mut, visit_relations,
 };
-use datafusion::sql::sqlparser::dialect::DuckDbDialect;
 use datafusion::sql::sqlparser::parser::Parser;
 use datafusion::sql::sqlparser::tokenizer::Span;
-use datafusion_table_providers::duckdb::sql_table::DuckSqlExec;
-use datafusion_table_providers::sql::db_connection_pool::dbconnection::duckdbconn::DuckDBParameter;
+use datafusion_expr::sqlparser::ast::GroupByExpr;
 use datafusion_table_providers::util::column_reference::ColumnReference;
 use datafusion_table_providers::util::indexes::IndexType;
-use duckdb::DuckdbConnectionManager;
-use r2d2::PooledConnection;
 use std::collections::HashSet;
 use std::fmt::Debug;
 use std::ops::ControlFlow;
 use std::sync::Arc;
 
-static DIALECT: DuckDbDialect = DuckDbDialect {};
 const CTE_NAME: &str = "_intermediate_materialize";
 
 pub struct DuckDBIntermediateIndexMaterializationOptimizer {}
 
 #[derive(Debug, Clone, PartialEq)]
 struct SelectionWithIdents {
     expr: Expr,
-    references: HashSet<String>,
+    references: HashSet<Expr>,
 }
 
 impl SelectionWithIdents {
     pub fn from(expr: &Expr) -> Self {
         let mut references = HashSet::new();
 
         let _ = visit_expressions(expr, |e| {
-            if let Expr::Identifier(id) = e {
-                references.insert(id.value.clone());
+            if let ident @ (Expr::Identifier(_) | Expr::CompoundIdentifier(_)) = e {
+                references.insert(ident.clone());
+            } else { /* no-op */
             }
```

Comment:
```suggestion
}
```

remove the no op branch?